lib/broadway_cloud_pub_sub/pull_client.ex

```diff
@@ -171,8 +171,7 @@ defmodule BroadwayCloudPubSub.PullClient do
     %{message | "data" => Base.decode64!(encoded_data)}
   end
 
-  defp decode_message(%{"data" => nil} = message), do: message
-  defp decode_message(%{"attributes" => %{"payloadFormat" => "NONE"}} = message), do: message
+  defp decode_message(message) when is_map(message) and not is_map_key(message, "data"), do: message
```

Comment:
Maybe deleting the case where `data` is in the map, but its value is `nil`, was a bit over-eager?

```suggestion
  defp decode_message(%{"data" => nil} = message), do: message
  defp decode_message(message) when is_map(message) and not is_map_key(message, "data"), do: message
```